scripts/ai/update-metadata.ts

```diff
@@ -83,10 +107,29 @@ async function processFile(file: string) {
     };
 
     if (offeringId) {
-      let [_full, unitCode, investigation, problem] = offeringId.match(/(.*)(\d)(\d\d)/);
+      // Extract the unit, investigation, and problem from the offeringId.
+      // The `offeringId` structure can vary. In some cases, there is no unit code. There are also cases where
+      // there is no investigation number. For example, in demo mode if the unit is not specified, there will
+      // be no unit value. And in the case where the investigation is 0 (like with the Intro to CLUE
+      // investigation in the Introduction to CLUE unit) the investigation will be undefined. In those cases,
+      // we default to "sas" for the unit and "0" for the investigation.
+      let unitCode = "";
+      let investigation = "";
+      let problem = "";
+      const match = offeringId.match(/(.*?)(\d)(\d\d)$/);
+
+      if (match) {
+        [, unitCode, investigation, problem] = match;
+      } else {
+        investigation = "0";
+        problem = offeringId.match(/\d+/)?.[0] || "";
+      }
```

Comment:
@scytacki Lucy and I saw a few cases where because the Introduction to CLUE unit was used by a student, the `offeringId` was set to just the problem number (e.g. "1"). We weren't sure how often that will be an issue and if we really need to record this information or skip it?

Reply:
I think it will be a common issue, so I think the way you handled it is good.